### PR TITLE
add proposal for Flux at Google Season of Docs 2023

### PR DESCRIPTION
**Pull request by dholbach**
I tried to follow [the template](https://developers.google.com/season-of-docs/docs/org-proposal-template) and took inspiration from other applications. Please let me know if anything is missing or could be explained better, also if the project scope is too wide/narrow or the budget idea is off. We appreciate advice from folks who participated in the project before - it's our first time! :-)

### CODE REVIEW

**Comment by nate-double-u** [comment]
Thanks for the proposal! 

We are looking at the best way to manage several proposals with the GSoD program (https://groups.google.com/g/season-of-docs-discuss/c/PxTWKQfblLc). We'll keep you updated and review this proposal regardless of how GSoD suggests we proceed.

**Comment by nate-double-u** [comment]
> We appreciate advice from folks who participated in the project before - it's our first time! :-)

This is my first time working with this program too 🙂
We'll learn together!

File: seasonofdocs/flux.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+## Improving the Flux User Onramp
+
+### About Flux
+
+[Flux](https://fluxcd.io) is a tool for keeping Kubernetes clusters in sync with sources of configuration (like Git repositories and OCI artifacts), and automating updates to configuration when there is new code to deploy.
+
+Flux was created in 2016 and helped to formalise GitOps, or "Operations by Pull Request" as industry standard. In 2020 a rewrite of the project was started which helped to drastically modernise the code base and make new use-cases possible.
+
+Flux is a graduated Cloud Native Computing Foundation (CNCF) project, used in production by various organisations and cloud providers.
```
Comment (aliok) [inline seasonofdocs/flux.md:18]:
```suggestion
Flux is licensed under the Apache License 2.0 license and is a graduated Cloud Native Computing Foundation (CNCF) project, used in production by various organisations and cloud providers.
```

File: seasonofdocs/flux.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+- [Information about the initiative](https://developers.google.com/season-of-docs/docs)
+- Current status of proposal: **Draft**
+
+### [Timeline](https://developers.google.com/season-of-docs/docs/timeline)
+
+## Improving the Flux User Onramp
+
+### About Flux
+
+[Flux](https://fluxcd.io) is a tool for keeping Kubernetes clusters in sync with sources of configuration (like Git repositories and OCI artifacts), and automating updates to configuration when there is new code to deploy.
```
Comment (aliok) [inline seasonofdocs/flux.md:14]: Maybe mention why this is important for users? It makes their lives easier? What does it increase like productivity/performance/throughput (I am just making up)?

**Comment by nate-double-u** [comment]
As per [the Google PMs comments](https://groups.google.com/g/season-of-docs-discuss/c/PxTWKQfblLc/m/nBrC8CRuAQAJ) about org applications:

> We still ask that each project apply separately; if a project is selected, they can choose CNCF as their payment option with Open Collective when we begin setting up grant payments.
> 
> Projects can feel free to add in their application that payments would be handled via CNCF but that's not a factor in our decision making for what projects to accept.

We ask that you submit this project directly with GSoD, rather than going through us (the CNCF). Sorry for the confusion around this @dholbach.

**Review by aliok — COMMENTED** [review]
thanks for the proposal @dholbach 

I've added some comments.

I am not super experienced with GSoD, but I checked 2 successful applications and did some comparison:
- https://github.com/cilium/cilium/blob/f110b95dc021d747ad6d9e3b1c3f76a31ca46c88/Documentation/contributing/development/season_of_docs.rst
- https://wiki.openmrs.org/display/RES/2022+Google+Season+of+Docs

First thing I noticed is that they are following the [template](https://developers.google.com/season-of-docs/docs/org-proposal-template) almost 100%.

I think it is a good idea.

Also, at this stage, Flux community would be convincing Google to give them the opportunity to hire a technical writer. So, what Google wants to see and how they want to see is important.

I would be happy to do a second round of reviews later on

**Comment by aliok** [comment]
> We ask that you submit this project proposal directly with GSoD, rather than going through us (the CNCF). Sorry for the confusion around this @dholbach.

@dholbach Even though the the application needs to happen individually by Flux community, I would be happy to review the proposal again, if you would like me to.

File: seasonofdocs/flux.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+- Review Installation docs - should they be broken up? Are references between these and others docs missing, e.g. the cheatsheets and CLI reference?
+- Background information: [#523](https://github.com/fluxcd/website/issues/523)
+
+#### Measuring the success
+
+As the Flux developers spend quite a bit of their time on supporting Flux users, and sometimes it takes a bit of time to discover the misunderstanding of a new user, we ideally would like to see the number of support requests go down and get the sense that users can more easily "help themselves".
+
+This will be hard to measure, especially as we are in a growing field and have a growing user-base. Still we want to monitor the number of Github discussions and Slack requests and compare with the previous months.
+
+#### Timeline
```
[the remark this answers is not in the document]
Comment (aliok) [reply seasonofdocs/flux.md:75]: UPDATE: I am not so sure if the project needs to be longer. Please ignore my comment about that. I would be happy to hear what you folks think about that matter. 

https://developers.google.com/season-of-docs/docs/admin guide#should_your_organization_participate_in_the_program

cc @dholbach @nate-double-u

[the remark this answers is not in the document]
Comment (nate-double-u) [reply seasonofdocs/flux.md:75]: It looks like the link you're sharing broke half way through. I'm not sure where you're intending to  link -- could you update @aliok?

**Comment by dholbach** [comment]
@nate-double-u @aliok Thanks so much for the reviews, I will dive into this today and send an update again!

File: seasonofdocs/flux.md [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+Flux is a graduated Cloud Native Computing Foundation (CNCF) project, used in production by various organisations and cloud providers.
+
+### Getting Started
+
+If you are interested in participating in Google Season of Docs as a technical writer, please do the following:
+
+1. Join the `#flux-contributors` channel on [CNCF Slack](http://slack.cncf.io/).
+1. Review this project proposal and familiarise yourself with the [documentation website](https://fluxcd.io/flux). Particularly the sections `Core Concepts`, `Get Started` and `Installation` are read by our users all them time. Getting a general sense of the structure of the organisation of the documentation will be of help as well.
+1. You will need some familiarity with Cloud Native technologies such as [Kubernetes](https://kubernetes.io), so you can more easily understand the problems and background of users.
+1. Talk to us on Slack, introduce yourself - we are happy to tell you more.
```
[the remark this answers is not in the document]
Comment (dholbach) [reply seasonofdocs/flux.md]: I'll move it to the end as you suggested. Somebody pointed out e.g. https://www.jenkins.io/sigs/docs/gsod/ as an example where there were clear instructions on how to find out more and get started.

**Comment by nate-double-u** [comment]
Closing this PR as this proposal has been published over on the fluxcd.io website.
https://fluxcd.io/contributing/docs/google-season-of-docs-2023/

Good luck @dholbach!

**Comment by nate-double-u** [comment]
I should say, that just because the proposal isn't kept in this repo, doesn't mean we're not following along! Please keep us up to date with how this goes - we're rooting for you!

**Comment by aliok** [comment]
@dholbach Great stuff!

Just had a quick look, it looks great! I hope your proposal gets accepted.

**Pull request closed without merging**